src/Runtime.Base/src/System/Runtime/RuntimeExports.cs

```diff
@@ -186,13 +181,10 @@ public static unsafe void RhUnboxAny(object o, ref byte data, EETypePtr pUnboxTo
         public static unsafe ref byte RhUnbox2(EETypePtr pUnboxToEEType, Object obj)
         {
             EEType* ptrUnboxToEEType = (EEType*)pUnboxToEEType.ToPointer();
-            if (obj.EEType != ptrUnboxToEEType)
+            if ((obj == null) || ((obj.EEType != ptrUnboxToEEType) && !UnboxAnyTypeCompare(obj.EEType, ptrUnboxToEEType)))
```

Comment:
It does not inline it if the basic block is rarely executed. But I agree that it may be premature optimization, especially considering how other JIT helpers are written. Once we get to working on runtime perf, we may need to go over all JIT helpers and fine tune their performance.